### PR TITLE
feat: Add `exception` opentype implementation to `Oprint`

### PR DESCRIPTION
**Pull request by spotandjake**
Fixes: #1894

Technically we may want to use another prefix compared to `exception` but given they are the only open type in grain. this should be good for now.

### CODE REVIEW

**Review by jozanza — APPROVED** [review]

**Review by peblair — COMMENTED** [review]
Code changes look good here, but I am missing the necessary context to know whether `exception` is what we want to print here. I'll let others take a peek at it before I approve.

File: compiler/src/typed/oprint.re [hunk 1]
```diff
@@ -767,7 +767,7 @@ and print_out_type_decl = (kwd, ppf, td) => {
         print_list(print_out_constr, ppf => fprintf(ppf, ",@;<0 2>")),
         constrs,
       )
-    | Otyp_open => fprintf(ppf, " = ..")
+    | Otyp_open => fprintf(ppf, " = ...")
```
Comment (phated) [inline compiler/src/typed/oprint.re:770]: Why did you change this to `...` instead of `..`?

**Review by phated — CHANGES_REQUESTED** [review]
Exception is our only open type, so that's correct. I'm not sure why you changed `..` to `...`

File: compiler/src/typed/oprint.re (hunk 1 above)
Comment (spotandjake) [reply compiler/src/typed/oprint.re]: I think that was a mistake. Its been corrected

**Review by phated — APPROVED** [review]
